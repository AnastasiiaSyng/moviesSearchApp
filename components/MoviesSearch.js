import React, {useState} from 'react';
import {styles} from './MoviesSearch.styles';
import {
  Image,
  Text,
  View,
  TextInput,
  ScrollView,
  Modal,
  Button,
  TouchableHighlight,
} from 'react-native';

import {API_KEY} from '@env';

export default function MoviesSearch() {
  const [state, setState] = useState({
    movies: '',
    moviesList: [],
    selectedMovie: {},
  });
  const {movies, moviesList, selectedMovie} = state;

  const getMoviesFromApi = () => {
    return fetch(
      `https://api.themoviedb.org/3/search/movie?api_key=${API_KEY}&language=en-US&query=${movies}&page=1`,
    )
      .then(response => response.json())
      .then(({results}) => {
        setState(prevState => {
          return {...prevState, moviesList: results};
        });
      })
      .catch(error => {
        console.error(error);
      });
  };
  const openSelectedMovie = id => {
    return fetch(
      `https://api.themoviedb.org/3/movie/${id}?api_key=${API_KEY}&language=en-US`,
    )
      .then(response => response.json())
      .then(data => {
        setState(prevState => {
          return {...prevState, selectedMovie: data};
        });
      })
      .catch(error => {
        console.error(error);
      });
  };

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Movie search</Text>
      <TextInput
        placeholder="Find movie..."
        style={styles.searchInput}
        value={movies}
        onChangeText={text =>
          setState(prevState => {
            return {...prevState, movies: text};
          })
        }
        onSubmitEditing={getMoviesFromApi}
      />

      <ScrollView style={styles.results}>
        {moviesList?.map(({id, backdrop_path, original_title}) => (
          <View key={id} style={styles.result}>
            <Image
              style={styles.image}
              source={{
                uri: 'https://image.tmdb.org/t/p/w500' + backdrop_path,
              }}
            />
            <Text style={styles.heading}>{original_title}</Text>
            <TouchableHighlight style={styles.buttonStyles}>
              <Button
                color="#374785"
                title="View"
                onPress={() => openSelectedMovie(id)}
              />
            </TouchableHighlight>
          </View>
        ))}
      </ScrollView>

      <Modal
        animationType="fade"
        transparent={false}
        visible={typeof selectedMovie.id !== 'undefined'}>
        <View style={styles.modalContainer}>
          <Image
            style={styles.imageModal}
            source={{
              uri:
                'https://image.tmdb.org/t/p/w500' + selectedMovie.backdrop_path,
            }}
          />
          <Text style={styles.infoModal}>{selectedMovie.original_title}</Text>
          <Text style={styles.infoModal}>{selectedMovie.overview}</Text>
          <Text style={styles.infoModal}>Budget:${selectedMovie.budget}</Text>
          <Text style={styles.infoModal}>
            Releas date: {selectedMovie.release_date}
          </Text>

          <TouchableHighlight style={styles.buttonStyles}>
            <Button
              color="#374785"
              title="Back"
              onPress={() =>
                setState(prevState => {
                  return {...prevState, selectedMovie: {}};
                })
              }
            />
          </TouchableHighlight>
        </View>
      </Modal>
    </View>
  );
}
