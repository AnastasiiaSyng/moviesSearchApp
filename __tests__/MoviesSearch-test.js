import React from 'react';
import MoviesSearch from '../components/MoviesSearch';
import renderer from 'react-test-renderer';

describe('MoviesSearch', () => {
  it('renders correctly MoviesSearch', () => {
    const tree = renderer.create(<MoviesSearch />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
