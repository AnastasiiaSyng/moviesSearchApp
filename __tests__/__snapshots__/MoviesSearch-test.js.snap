// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesSearch renders correctly MoviesSearch 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#374785",
      "flex": 1,
      "justifyContent": "flex-start",
      "opacity": 0.9,
      "paddingHorizontal": 10,
      "paddingTop": 50,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "Cochin",
        "fontSize": 35,
        "fontWeight": "bold",
        "marginBottom": 30,
        "textAlign": "center",
      }
    }
  >
    Movie search
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Find movie..."
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 5,
        "fontSize": 20,
        "fontWeight": "300",
        "marginBottom": 40,
        "padding": 20,
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#24305E",
        "marginBottom": 20,
        "width": "100%",
      }
    }
  >
    <View />
  </RCTScrollView>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8E9A1",
          "flex": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://image.tmdb.org/t/p/w500undefined",
          }
        }
        style={
          Object {
            "height": 250,
            "marginVertical": 40,
            "width": "100%",
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "Cochin",
            "fontSize": 20,
            "fontWeight": "200",
            "marginBottom": 10,
            "textAlign": "center",
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "Cochin",
            "fontSize": 20,
            "fontWeight": "200",
            "marginBottom": 10,
            "textAlign": "center",
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "Cochin",
            "fontSize": 20,
            "fontWeight": "200",
            "marginBottom": 10,
            "textAlign": "center",
          }
        }
      >
        Budget:$
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "Cochin",
            "fontSize": 20,
            "fontWeight": "200",
            "marginBottom": 10,
            "textAlign": "center",
          }
        }
      >
        Releas date: 
      </Text>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#F76C6C",
            "borderRadius": 10,
            "height": 40,
            "marginHorizontal": 5,
            "marginTop": 20,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#374785",
                  },
                ]
              }
            >
              Back
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
